Marlin/src/module/tool_change.cpp

```diff
@@ -494,225 +476,219 @@ void tool_change(const uint8_t tmp_extruder, const float fr_mm_s/*=0.0*/, bool n
-          #if ENABLED(ADVANCED_PAUSE_FEATURE)
-            do_pause_e_move(-toolchange_settings.swap_length, MMM_TO_MMS(toolchange_settings.retract_speed));
-          #else
-            current_position[E_AXIS] -= toolchange_settings.swap_length / planner.e_factor[active_extruder];
-            planner.buffer_line(current_position, MMM_TO_MMS(toolchange_settings.retract_speed), active_extruder);
+        #if ENABLED(SINGLENOZZLE)
+          #if FAN_COUNT > 0
+            singlenozzle_fan_speed[active_extruder] = fan_speed[0];
+            fan_speed[0] = singlenozzle_fan_speed[tmp_extruder];
           #endif
-        }
-
-        constexpr float snfr =
-          #if ENABLED(SINGLENOZZLE_SWAP_PARK)
-            MMM_TO_MMS(SINGLENOZZLE_PARK_XY_FEEDRATE);
-          #else
-            0
-          #endif
-        ;
-
-        float old_pos[XYZ];
 
-        if (!no_move) {
-          COPY(old_pos, current_position);
-
-          #if ENABLED(SINGLENOZZLE_SWAP_PARK)
-            current_position[X_AXIS] = toolchange_settings.change_point.x;
-            current_position[Y_AXIS] = toolchange_settings.change_point.y;
-          #endif
-          current_position[Z_AXIS] += toolchange_settings.z_raise;
-
-          do_blocking_move_to(current_position, snfr);
-        }
-
-        singlenozzle_temp[active_extruder] = thermalManager.target_temperature[0];
-        if (singlenozzle_temp[tmp_extruder] && singlenozzle_temp[tmp_extruder] != singlenozzle_temp[active_extruder]) {
-          thermalManager.setTargetHotend(singlenozzle_temp[tmp_extruder], 0);
-          #if ENABLED(ULTRA_LCD)
-            thermalManager.set_heating_message(0);
-          #endif
-          (void)thermalManager.wait_for_hotend(0, false);  // Wait for heating or cooling
-        }
-
-        active_extruder = tmp_extruder;
+          singlenozzle_temp[active_extruder] = thermalManager.target_temperature[0];
+          if (singlenozzle_temp[tmp_extruder] && singlenozzle_temp[tmp_extruder] != singlenozzle_temp[active_extruder]) {
+            thermalManager.setTargetHotend(singlenozzle_temp[tmp_extruder], 0);
+            #if ENABLED(ULTRA_LCD)
+              thermalManager.set_heating_message(0);
+            #endif
+            (void)thermalManager.wait_for_hotend(0, false);  // Wait for heating or cooling
+          }
+          active_extruder = tmp_extruder;
+        #endif
 
-        if (toolchange_settings.swap_length) {
-          #if ENABLED(ADVANCED_PAUSE_FEATURE)
-            do_pause_e_move(toolchange_settings.swap_length, toolchange_settings.prime_speed);
-          #else
-            current_position[E_AXIS] += toolchange_settings.swap_length / planner.e_factor[tmp_extruder];
-            planner.buffer_line(current_position, toolchange_settings.prime_speed, tmp_extruder);
+        #if ENABLED(TOOLCHANGE_FILAMENT_SWAP)
+          #if ENABLED(PREVENT_COLD_EXTRUSION)
+            if (!DEBUGGING(DRYRUN) && thermalManager.targetTooColdToExtrude(active_extruder)) {
```

Comment:
This logic looks exactly backwards: "If too_cold then go ahead and do the swap?"